File: sword-means-offer-2/006/main.go
package main

import "fmt"

// @Time     : 2021/12/14 22:35
// @Author   : Ranshi
// @File     : main.go
// @Doc      : 剑指 Offer II 006. 排序数组中两个数字之和

func twoSum(numbers []int, target int) []int {
	left, right := 0, len(numbers)-1
	idx := 0
	for left < right {
		idx = numbers[left] + numbers[right]
		if idx > target {
			right--
		} else if idx < target {
			left++
		} else {
			return []int{left, right}
		}
	}
	return nil
}

func main() {
	fmt.Println(twoSum([]int{1, 2, 4, 6, 10}, 8))
}

File: sword-means-offer-2/006/cpp/main.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
  vector<int> twoSum(vector<int> &numbers, int target) {
    for (int i = 0, j = numbers.size() - 1; i < j;) {
      if (numbers[i] + numbers[j] < target)
        ++i;
      else if (numbers[i] + numbers[j] > target)
        --j;
      else
        return {i, j};
    }
    return {-1, -1};
  }
};

int main(int argc, char const *argv[]) {
  vector<int> numbers = {1, 2, 4, 6, 10};
  auto res = Solution().twoSum(numbers, 8);
  for (auto num : res) {
    cout << num;
  }
  return 0;
}
